### PR TITLE
feat: enable hermes

### PR DESCRIPTION
**Pull request by gkartalis**
This PR resolves [MOPLAT-577] 

### Description

Enabling hermes on [android only](https://reactnative.dev/docs/0.67/native-modules-android)

available on android beta [android-8.11.0-1674645224](https://github.com/artsy/eigen/releases/tag/android-8.11.0-1674645224)

Bonus wins:

- Hermes debugger on flipper android 🔥
- aab file size seems to be smaller (89.99MB with hermes vs 110.57 without hermes)

#### Next steps:

We can test it out with applause to include it on the next release (?) after we QA ourselves

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

- enables hermes on android - gkartalis

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[MOPLAT-577]: https://artsyproduct.atlassian.net/browse/MOPLAT-577?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: android/app/proguard-rules.pro [hunk 1]
```diff
@@ -8,3 +8,7 @@
 #   http://developer.android.com/guide/developing/tools/proguard.html
 
 # Add any project specific keep options here:
+
+# needed for hermes
+-keep class com.facebook.hermes.unicode.** { *; }
+-keep class com.facebook.jni.** { *; }
```
Comment (gkartalis) [inline android/app/proguard-rules.pro:14]:
```suggestion
-keep class com.facebook.jni.** { *; }

```

File: package.json [hunk 2]
```diff
@@ -217,6 +217,7 @@
     "@babel/core": "7.16.12",
     "@babel/plugin-proposal-decorators": "7.18.9",
     "@babel/plugin-transform-flow-strip-types": "7.18.9",
+    "@babel/plugin-transform-named-capturing-groups-regex": "7.20.5",
```
Comment (gkartalis) [inline package.json:220]: This is needed, without it we get [this crash](https://artsynet.sentry.io/issues/4059089208/?project=5867225&query=is%3Aunresolved+is%3Afor_review+assigned_or_suggested%3A%5Bme%2C+none%5D&referrer=issue-stream&sort=date&statsPeriod=14d&stream_index=0)

resolves the issue with hermes and regex

**Comment by gkartalis** [comment]
one weird thing that I haven't figured out is that for some reason the NativeModules from "react-native" is an empty object (the only surface it affects is our devMenu -> can't open the react native dev menu but I wouldn't consider this launch blocking

**Review by damassi — DISMISSED** [review]
Really happy this is moving again! 👍 👍

**Comment by damassi** [comment]
> can't open the react native dev menu 

@gkartalis - Maybe add a follow-up JIRA ticket so that this isn't lost?

**Comment by admbtlr** [comment]
Doesn't this impact all the other areas of the app that use `NativeModules`?

**Comment by gkartalis** [comment]
> Doesn't this impact all the other areas of the app that use NativeModules?
No it doesn't! We double checked with @brainbicycle today and it looks safe!

We can come up with a testing plan but it already looks like a huge improvement,

Hermes debugger works SO GOOD on flipper and also it never crashes so far (even on the broken stickyTab stuff that we are not able to debug with RNDebugger)

I AM SO EXCITED 🤩

**Pull request merged**